lib/queue.rb

```diff
@@ -1,31 +1,59 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(30)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+    
+    if !@store[@back].nil? && @front == @back
+      raise StandardError.new('queue is full.')
+    end
+    
+    @store[@back] = element
+    @back = (@back+1) % @store.length
+    
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    dequeued = @store[@front]
+    
+    # remove dequeued item from queue
+    @store[@front] = nil
+    # move front to next item in queue
+    @front = @front+1
+    
+    return dequeued    
   end
-
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.compact.length
   end
-
+  
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    if self.size == 0
+      return true
+    else
+      return false
+    end
   end
-
+  
+  # helper method for tests
+  def length 
+    return @store.length
+  end
+  
   def to_s
+    @store.compact!
```

Comment:
This compacts the buffer which removes any extra space you have in the buffer, which you might need in the Queue!  

Better to do something like this:

```ruby
    return "[]" if @front == -1
    current = @front
    output = "["
    next = (@front + 1) % @store.length
    while next != @back
      output += "#{@store[current]},"
      current = (current + 1) % @store.length
      next = (next + 1) % @store.length
    end

    output += "#{@store[current]}]"
    return output
```